File: BoardGameGeek/src/com/boardgamegeek/provider/GamesCategoriesIdProvider.java
package com.boardgamegeek.provider;

import android.content.ContentUris;
import android.net.Uri;
import android.provider.BaseColumns;

import com.boardgamegeek.provider.BggContract.Categories;
import com.boardgamegeek.provider.BggDatabase.Tables;
import com.boardgamegeek.util.SelectionBuilder;

public class GamesCategoriesIdProvider extends BaseProvider {

	@Override
	protected SelectionBuilder buildSimpleSelection(Uri uri) {
		long id = ContentUris.parseId(uri);
		return new SelectionBuilder().table(Tables.GAMES_CATEGORIES).whereEquals(BaseColumns._ID, id);
	}

	@Override
	protected String getPath() {
		return "games/categories/#";
	}

	@Override
	protected String getType(Uri uri) {
		return Categories.CONTENT_ITEM_TYPE;
	}
}

File: app/src/main/java/com/boardgamegeek/provider/GamesCategoriesIdProvider.kt
package com.boardgamegeek.provider

import android.content.ContentUris
import android.net.Uri
import android.provider.BaseColumns
import com.boardgamegeek.provider.BggContract.Categories
import com.boardgamegeek.provider.BggContract.Companion.PATH_CATEGORIES
import com.boardgamegeek.provider.BggContract.Companion.PATH_GAMES
import com.boardgamegeek.provider.BggDatabase.Tables

class GamesCategoriesIdProvider : BaseProvider() {
    override fun getType(uri: Uri) = Categories.CONTENT_ITEM_TYPE

    override val path = "$PATH_GAMES/$PATH_CATEGORIES/#"

    override val defaultSortOrder = Categories.DEFAULT_SORT

    override fun buildSimpleSelection(uri: Uri): SelectionBuilder {
        val id = ContentUris.parseId(uri)
        return SelectionBuilder().table(Tables.GAMES_CATEGORIES).whereEquals(BaseColumns._ID, id)
    }
}
